modules/gltf/src/lib/extensions/KHR_texture_transform.ts

```diff
@@ -0,0 +1,194 @@
+/**
+ * https://github.com/KhronosGroup/glTF/blob/main/extensions/2.0/Khronos/KHR_texture_transform/README.md
+ */
+
+import {Vector3, Matrix3} from '@math.gl/core';
+import type {GLTF, GLTFMaterial, GLTFMeshPrimitive} from '../types/gltf-types';
+import type {GLTFLoaderOptions} from '../../gltf-loader';
+import {BYTES, COMPONENTS} from '../api/post-process-gltf';
+import {getAccessorArrayTypeAndLength} from '../gltf-utils/gltf-utils';
+import {TypedArray} from '@loaders.gl/schema';
+
+/** Extension name */
+const EXT_MESHOPT_TRANSFORM = 'KHR_texture_transform';
+
+export const name = EXT_MESHOPT_TRANSFORM;
+
+const scratchVector = new Vector3();
+const scratchMatrix = new Matrix3();
+
+export async function decode(
+  gltfData: {
+    json: GLTF;
+    buffers: {arrayBuffer: ArrayBuffer; byteOffset: number; byteLength: number}[];
+  },
+  options: GLTFLoaderOptions
+) {
+  const materials = gltfData.json.materials || [];
+  for (let i = 0; i < materials.length; i++) {
+    const material = materials[i];
+    const extensionData = getExtensionData(material);
+    if (extensionData) {
+      const matrix = makeTransformationMatrix(extensionData);
+      transformPrimitives(gltfData, i, matrix);
+    }
+  }
+}
+
+function getExtensionData(material: GLTFMaterial): any {
+  const baseColorTexture = material.pbrMetallicRoughness?.baseColorTexture;
+  if (baseColorTexture) {
+    return baseColorTexture.extensions?.[EXT_MESHOPT_TRANSFORM];
```

Comment:
Can we just pick one of these? Don't we need to apply separately to all of them? Couldn't they have different transforms?